### PR TITLE
feat-sonarqube: Added sonar quality gateway conversion

### PR DESCRIPTION
**Pull request by vanisrikanithi**
Sonar Quality Gate go-conversion:
Added "Sonar Quality Gate" support to go-convert specific to jenkins migration.

Sonar Quality Gate Jenkins Trace:
[vani_sonar_Trace.json](https://github.com/user-attachments/files/17926111/vani_sonar_Trace.json)

go run main.go jenkinsjson --downgrade ~/go/vani_sonar_Trace.json

Sonar Quality Gate Harness test pipeline [here](https://app.harness.io/ng/account/gCoPSwHxS7ipOgx2iA9tOQ/module/ci/orgs/default/projects/jenkinstoharness/pipelines/Vani_Sonarqube/deployments/MtoXJuhmQ3aI2M_1KDesyw/pipeline?storeType=INLINE).

Sonar Quality Gate harness pipeline step:
```
- step:
    identifier: waitforqualitygate6278c9
    name: Sonarqube_Quality_Gate
    spec:
      image: 'plugins/sonarqube-scanner:v2.4.2'
      settings:
        scm_disabled: 'false'
        skip_scan: 'false'
        sonar_host: <+input>
        sonar_key: <+input>
        sonar_name: <+input>
        sonar_organization: <+input>
        sonar_quality_enabled: 'true'
        sonar_qualitygate: OK
        sonar_token: <+input>
        sources: .
        timeout: '300'
    timeout: ''
    type: Plugin
```

### CODE REVIEW

**Comment by vanisrikanithi** [comment]
Hello @Ompragash,

Could you please review the PR for sonar quality gate conversion and provide your comments on that?

Thank you.

**Review by Ompragash — CHANGES_REQUESTED** [review]
Tests failed with below as the test files related to existing implementation of withSonarQubeEnv is removed
```
--- FAIL: TestConvertSonarqube (0.00s)
    convert_test.go:275: failed to read JSON file: open /drone/src/convert/jenkinsjson/convertTestFiles/sonarqube/sonarqubeSnippet.json: no such file or directory
FAIL
```

File: convert/jenkinsjson/json/sonarQualityGate.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+// ConvertSonarQualityGate creates a Harness step for nunit plugin.
+func ConvertSonarQualityGate(node Node) *harness.Step {
+
+	convertSonarQualityGate := &harness.Step{
+		Name: "Sonarqube_Quality_Gate",
+		Type: "plugin",
+		Id:   SanitizeForId(node.SpanName, node.SpanId),
+		Spec: &harness.StepPlugin{
+			Image: "plugins/sonarqube-scanner:v2.4.2",
```
Comment (Ompragash) [inline convert/jenkinsjson/json/sonarQualityGate.go:15]:
```suggestion
			Image: "plugins/sonarqube-scanner:latest",
```

File: convert/jenkinsjson/json/sonarQualityGate_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+func TestConvertSonarQualityGate(t *testing.T) {
+
+	var tests []runner
+	tests = append(tests, prepare(t, "/sonarQualityGate/sonarQualityGate_snippet", &harness.Step{
+		Id:   "waitForQualityGate6278c9",
+		Name: "Sonarqube_Quality_Gate",
+		Type: "plugin",
+		Spec: &harness.StepPlugin{
+			Image: "plugins/sonarqube-scanner:v2.4.2",
```
Comment (Ompragash) [inline convert/jenkinsjson/json/sonarQualityGate_test.go:18]:
```suggestion
			Image: "plugins/sonarqube-scanner:latest",
```

File: convert/jenkinsjson/convertTestFiles/sonarQualityGate/sonarQuanlityGate.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+- step:
+    identifier: waitforqualitygate6278c9
+    name: Sonarqube_Quality_Gate
+    spec:
+      image: 'plugins/sonarqube-scanner:v2.4.2'
```
Comment (Ompragash) [inline convert/jenkinsjson/convertTestFiles/sonarQualityGate/sonarQuanlityGate.yaml:5]:
```suggestion
      image: 'plugins/sonarqube-scanner:latest'
```

**Review by Ompragash — APPROVED** [review]

**Review by Ompragash — APPROVED** [review]

**Pull request merged**